core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -824,6 +826,13 @@ public NativeImageInvokerInfo build() {
                 if (nativeConfig.enableVmInspection) {
                     nativeImageArgs.add("-H:+AllowVMInspection");
                 }
+                if (nativeConfig.monitoring.isPresent()) {
+                    List<NativeConfig.MonitoringOption> monitoringOptions = nativeConfig.monitoring.get();
+                    if (!monitoringOptions.isEmpty()) {
+                        nativeImageArgs.add("--enable-monitoring=" + monitoringOptions.stream().map(o -> o.name().toLowerCase(
+                                Locale.ROOT)).collect(Collectors.joining(",")));
+                    }
```

Comment:
```suggestion
                    if (monitoringOptions.isEmpty()) {
                        nativeImageArgs.add("--enable-monitoring");
                    } else {
                        nativeImageArgs.add("--enable-monitoring=" + monitoringOptions.stream().map(o -> o.name().toLowerCase(
                                Locale.ROOT)).collect(Collectors.joining(",")));
                    }
```

[--enable-monitoring can be passed without arguments as well](https://github.com/oracle/graal/commit/d5bd26a2211738dfea60a4d4f64343467fe4567f) (implies "all")

Reply:
I need to see if this works

Reply:
PR updated